plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -349,32 +347,33 @@ def get_metadata(self, metadata_files):
                     identified and downloaded.
         :rtype:     pulp_rpm.plugins.importers.yum.repomd.metadata.MetadataFiles
         """
-
         self.downloader = metadata_files.downloader
         scratchpad = self.conduit.get_scratchpad() or {}
         previous_revision = scratchpad.get(constants.REPOMD_REVISION_KEY, 0)
         previous_skip_set = set(scratchpad.get(constants.PREVIOUS_SKIP_LIST, []))
         current_skip_set = set(self.config.get(constants.CONFIG_SKIP, []))
         self.current_revision = metadata_files.revision
-        last_sync = self.conduit.last_sync()
-        sync_due_to_unit_removal = False
-        if last_sync is not None:
-            last_sync = dateutils.parse_iso8601_datetime(last_sync)
-            last_removed = self.repo.last_unit_removed
-            sync_due_to_unit_removal = last_removed is not None and last_sync < last_removed
         # determine missing units
         missing_units = repo_controller.missing_unit_count(self.repo.repo_id)
-        # if the current MD revision is not newer than the old one
+
+        force_full_sync = repo_controller.check_perform_full_sync(self.repo.repo_id,
+                                                                  self.conduit,
+                                                                  self.config)
+
+        # if the platform does not prescribe forcing a full sync
+        # the current MD revision is not newer than the old one
         # and we aren't using an override URL
         # and the skip list doesn't have any new types
         # and there are no missing units, or we have deferred download enabled
         # and no units were removed after last sync
         # then skip fetching the repo MD :)
-        if not self.force_full and 0 < metadata_files.revision <= previous_revision \
-                and not self.config.override_config.get(importer_constants.KEY_FEED) \
-                and previous_skip_set - current_skip_set == set() \
-                and (self.download_deferred or not missing_units) \
-                and not sync_due_to_unit_removal:
+        skip_sync = not force_full_sync and \
+            0 < metadata_files.revision <= previous_revision and \
+            not self.config.override_config.get(importer_constants.KEY_FEED) and \
+            previous_skip_set - current_skip_set == set() \
```

Comment:
i am not sure, but i think even the skip_set check can be removed, because 'type_skip_list' is part of importer config, and if we change the skip list that means we change the importer_config and if the config was changed since last sync then we do force_sync no matter what. That said, i think we will not arrive till this check because `force_full_sync` will be True.